alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureWorkflowRestApiIT.kt

```diff
@@ -0,0 +1,1406 @@
+            assertEquals("Delete workflow failed", RestStatus.OK, response?.restStatus())
+        } finally {
+            deleteRoleAndRoleMapping(TEST_HR_ROLE)
+            deleteUser(deleteUser)
+            deleteUserClient?.close()
+        }
+    }
+
+    fun `test execute workflow with an user with execute workflow access`() {
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            listOf(TEST_HR_BACKEND_ROLE),
+            getClusterPermissionsFromCustomRole(ALERTING_EXECUTE_WORKFLOW_ACCESS)
+        )
+
+        val monitor = createRandomMonitor(true)
+        val workflow = createRandomWorkflow(listOf(monitor.id), true)
+
+        try {
+            val executeWorkflowResponse = userClient?.makeRequest(
+                "POST",
+                "$WORKFLOW_ALERTING_BASE_URI/${workflow.id}/_execute",
+                mutableMapOf()
+            )
+            assertEquals("Executing workflow failed", RestStatus.OK, executeWorkflowResponse?.restStatus())
+        } finally {
+            deleteRoleAndRoleMapping(TEST_HR_ROLE)
+        }
+    }
+
+    fun `test execute workflow with an user without execute workflow access`() {
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            listOf(TEST_HR_BACKEND_ROLE),
+            getClusterPermissionsFromCustomRole(ALERTING_NO_ACCESS_ROLE)
+        )
+
+        val monitor = createRandomMonitor(true)
+        val workflow = createRandomWorkflow(listOf(monitor.id), true)
+
+        try {
+            userClient?.makeRequest(
+                "POST",
+                "$WORKFLOW_ALERTING_BASE_URI/${workflow.id}/_execute",
+                mutableMapOf()
+            )
+            fail("Expected 403 Method FORBIDDEN response")
+        } catch (e: ResponseException) {
+            assertEquals("Execute workflow failed", RestStatus.FORBIDDEN, e.response.restStatus())
+        } finally {
+            deleteRoleAndRoleMapping(TEST_HR_ROLE)
+        }
+    }
+
+    fun `test delete workflow with an user with delete workflow access`() {
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            listOf(TEST_HR_BACKEND_ROLE),
+            getClusterPermissionsFromCustomRole(ALERTING_DELETE_WORKFLOW_ACCESS)
+        )
+
+        val monitor = createRandomMonitor(true)
+        val workflow = createRandomWorkflow(monitorIds = listOf(monitor.id))
+        val refresh = true
+
+        try {
+            val deleteWorkflowResponse = userClient?.makeRequest(
+                "DELETE",
+                "$WORKFLOW_ALERTING_BASE_URI/${workflow.id}?refresh=$refresh",
+                emptyMap(),
+                monitor.toHttpEntity()
+            )
+            assertEquals("DELETE workflow failed", RestStatus.OK, deleteWorkflowResponse?.restStatus())
+        } finally {
```

Comment:
If we are doing the try and finally, we need to have a catch for any exception and fail on that. Or we dont do the try finally part. This is same with the other tests like this.